### ISSUE TITLE
RtspCameraExample issue when client is Hikvision NVR

### ISSUE THREAD

**Issue by carpediem-av** [ngraziano/SharpRTSP#84]
The RtspCameraExample works fine with VLC, but the Hikvision NVR with the server link added does not play the stream. The console shows that packets are sent, and then the connection is terminated, and so on in a loop. I wanted to use this program to display a static image on the DVR as if it were a real camera.

**Comment by ngraziano** [comment]
I do not have access to this kind of material. no way for me to test.

**Comment by bentorkington** [comment]
This is possibly related to #127. @carpediem-av, can you please use Wireshark to capture the conversation between Hikvision and RtspCameraExample, and attach it here. Please filter only the RTSP packets, similar to my example in #127.

**Comment by carpediem-av** [comment]
Yes, sure. 192.168.10.107 is PC.
[hik.zip](https://github.com/user-attachments/files/16802715/hik.zip)

**Comment by ngraziano** [comment]
I have add the sending of RTCP RR message in the example, it may help. Can you test ?

If it is not working, can you make a capture with a working client (VLC for exemple).

**Comment by carpediem-av** [comment]
We were talking about RtspCameraExample. Looks like you made the changes in the wrong place (RtspClientExample)

**Comment by ngraziano** [comment]
Sorry, I was totally wrong. I don't know why I was thinking you were using the program as client. As server, the RR can't be the problem.

In the capture, it seems that the NVR is connecting two time to the server. Maybe the NVR is not recognizing the stream (it is a special not compressed h264 stream) and it retry. 
Can you try configure the NVR to request stream in UDP to see if it change something ?

(for simulating camera you can also use the RSTP server happytime, it work well to stream a jpg or a video)

**Comment by carpediem-av** [comment]
No, it doesn't work via UDP either. VLC works great. I tried to encode the image through the libx264 library (integrated into the code), also to no avail (which means it’s not the encoder).

**Comment by RogerHardiman** [comment]
I have an old Hik NVR in the office. It connects to SharpRTSP.
Mine is a 7716 with firmware from 2022 (V4.61.025 build 220905)
![image](https://github.com/user-attachments/assets/f22ef6fe-5deb-4b96-bfef-22c96470e0e3)

What model and firmware have you got?
